tokio-util/src/task/spawn_aborting.rs

```diff
@@ -0,0 +1,38 @@
+use std::ops::{Deref, DerefMut};
+
+use tokio::task::JoinHandle;
+
+use futures_core::Future;
+
+/// This is a wrapper type around JoinHandle that allows it to be dropped.
+#[derive(Debug)]
+pub struct DropHandle<T>(JoinHandle<T>);
+
+impl<T> Drop for DropHandle<T> {
+    fn drop(&mut self) {
+        self.0.abort();
+    }
+}
+
+impl<T> Deref for DropHandle<T> {
+    type Target = JoinHandle<T>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<T> DerefMut for DropHandle<T> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// This function spawns a task that aborts on drop instead of lingering.
+pub fn spawn_aborting<F>(future: F) -> DropHandle<F::Output>
+where
+    F: Future + Send + 'static,
+    F::Output: Send + 'static,
+{
+    DropHandle(tokio::spawn(future))
+}
```

Comment:
I don't like this name. It sounds like something that would immediately make something abort.

How about `spawn_with_drop_handle` or `spawn_with_abort_handle`?

Reply:
Another possibility is to make this a constructor method. Then you would type `DropHandle::spawn`.

Reply:
For API naming, how about using `spawn_abortable` which aligns with [Abortable](https://docs.rs/futures/latest/futures/future/struct.Abortable.html) in future crate.